experimental/cloudstrapper/README.md

```diff
@@ -2,140 +2,178 @@
-       folder
+   - If you are using the Marketplace experience,
+     - Locate the Cloudstrapper AMI and version-specific AGW AMI and their respective AMI ids from Marketplace and make changes to cluster.yaml
+     - The AGW AMI id would be set to awsAgwAmi
+     - If using the Cloudstrapper tooling to provision the instance, the Cloudstrapper AMI id would be set to awsCloudstrapperAmi
+
+  - If you are building Magma by yourself or using your own repositories,
+
+    - Create private repo on github to host helm charts.
+     This information would be used in the build.yaml (buildHelmRepo) and cluster.yaml (gitHelmRepo) files when building and deploying Magma respectively.
+
+    - Update local secrets.yaml to include Github and Docker access information for your custom repo
+     - Github username and PAT (Personal Access Token)
+     - Dockerhub username and password
 
-## 1. Run aws-essentials to setup all AWS related components as a stack
+## 1. Run aws-essentials to setup all AWS related base components as a stack
 
   The aws-essentials playbook will:
   - Create boot and host keys if required using the keyCreate tag. Default is to not create keys.
   - Create security group on the default VPC
-  - Create default bucket for shared storage. Ensure bucket does not exist by checking defaults.yaml
-    under the 'bucketDefault' variable name
+  - Create default bucket for shared storage.
+  - Create default CloudFormation stack for all essential components
+  If the keys, security groups and the default bucket exist, this playbook can be skipped.
 
-  - Command:
-    ```
-    ansible-playbook aws-prerequisites.yaml -e 'awsTargetRegion=<< AWS Region >>' -e "dirInventory=<directory>" [ --tags keyCreate ]
-    ```
-  - Example:
-    ```
-    ansible-playbook aws-prerequisites.yaml -e 'awsTargetRegion=us-east-1' -e "dirInventory=~/magma-experimental/files" 
-    ``` 
-  - Result: Created stackMantleEssentials with common security group, S3 storage
+  Before running the playbook, validate the following variables in defaults.yaml and cluster.yaml in VARS_DIR
 
-### 1.1 For users who do not have access to a Cloudstrapper AMI: Optional CI/CD
-  The devops playbooks are used to initialize a default instance, configure it to act as a Cloudstrapper and generate an
-  AMI that can be used as Cloudstrapper AMI and either published in the Marketplace as a public or community AMI or 
-  retained locally.
+  - defaults.yaml
+    - secgroupDefault indicates name of the default SecurityGroup that would be created and used across instances
+    - bucketDefault indicates the name of the default S3 bucket used to persist information
+    - stackEssentialsDefault indicates the name of the CloudFormation stack that hosts all the artifacts
+    - keyBoot refers to the key used for the Cloudstrapper instance. Use the keyCreate tag or if it's an existing key, use the right name
```

Comment:
if keyCreate is used can this step be skipped ?

Reply:
The name of the keys generated via keyCreate is actually picked up from defaults.yaml. Once created, other playbooks use this information to instantiate running EC2 instances for Cloudstrapper and AGWs. This step of updating defaults.yaml to reflect the name of the keys cannot be skipped. Added text to reflect that if two keys already exist and user will be using those keys, updating defaults.yaml to reflect those names and skipping keyCreate is adequate.